frameworks/react/__tests__/AdvancedPicture.test.tsx

```diff
@@ -98,10 +101,183 @@ describe('AdvancedPicture', () => {
     setTimeout(() => {
       expect(component.html()).toContain(
         '<picture>' +
+        '<img src="https://res.cloudinary.com/demo/image/upload/sample"><source media="(max-width: 500px)" sizes="80vw" ' +
+        'srcset="https://res.cloudinary.com/demo/image/upload/c_crop,w_500/dog"></picture>');
+      done();
+    }, 0);// one tick
+  });
+
+  it('autoOptimalBreakpoints: should generate default srcset when min-width and max-width not provided',
+    function (done) {
+      const component = mount(
+        <AdvancedPicture
+          autoOptimalBreakpoints
+          cldImg={defualtImage}
+          sources={[
+            {
+              image: sepiaImage
+            }
+          ]}
+        />);
+      setTimeout(() => {
+        expect(component.html()).toContain('<picture>' +
+        '<img src="https://res.cloudinary.com/demo/image/upload/sample"><source sizes="100vw" ' +
+        'srcset="https://res.cloudinary.com/demo/image/upload/e_sepia/c_scale,w_828/sample 828w,' +
+        'https://res.cloudinary.com/demo/image/upload/e_sepia/c_scale,w_1366/sample 1366w,' +
+        'https://res.cloudinary.com/demo/image/upload/e_sepia/c_scale,w_1536/sample 1536w,' +
+        'https://res.cloudinary.com/demo/image/upload/e_sepia/c_scale,w_1920/sample 1920w,' +
+        'https://res.cloudinary.com/demo/image/upload/e_sepia/c_scale,w_3840/sample 3840w">' +
+        '</picture>'
+        );
+        done();
+      }, 0);// one tick
+    });
+
+  it('autoOptimalBreakpoints: should generate single src when min-width > max-width', function (done) {
+    const component = mount(
+      <AdvancedPicture
+        cldImg={defualtImage} sources={[
+          {
+            maxWidth: 500,
+            minWidth: 700,
+            image: smallImage
+          }
+        ]}
+      />);
+    setTimeout(() => {
+      expect(component.html()).toContain(
+        '<picture><img src="https://res.cloudinary.com/demo/image/upload/sample">' +
+        '<source media="(min-width: 700px) and (max-width: 500px)" sizes="100vw" ' +
+        'srcset="https://res.cloudinary.com/demo/image/upload/c_crop,w_500/dog"></picture>'
```

Comment:
This specific configuration means nothing will ever be shown.
The media query can never be true here. `X > 700 && X < 500`

Reply:
@patrick-tolosa what do you think the expected behaviour should be in this case

Reply:
@const-cloudinary are you okay with going in that direction? I'll update the spec if that's the case

Reply:
@const-cloudinary Discussed with @patrick-tolosa and decided to not have a special behaviour for this case and console log error